### ISSUE TITLE
Cant upload large .gpx

### ISSUE THREAD

**Issue by beretor** [ThomasParistech/pretty-gpx#2]
Hello

Super nice project. 

I cant upload a large .gpx (25Mo). I suspect this is the expected behaviour but there is little info about it.
(There is a small warning icon next to the file but no clear info about it)

<img width="355" alt="Screenshot 2024-09-11 at 15 41 13" src="https://github.com/user-attachments/assets/266cdb00-56f8-42a7-863d-8b47fb8074e1">

**Comment by ThomasParistech** [comment]
I kept the `max_total_size` property of the drag&drop nicegui.upload element to None, so it shouldn't be a problem. But sometimes it fails to upload the file... Most of the time, clearing it and feeding it again works, give it a try!

I should add a clearer warning about it (or write a github issue on nicegui's repo)

Can you send me the GPX ?

**Comment by beretor** [comment]
Understood.
I tried a few times and it still fails. it doesnt even start.

here is the gpx: https://drive.google.com/file/d/1y4Du9VE8jJe6BkkTUvoo9Ohgzueo7QtR/view?usp=drive_web

**Comment by ThomasParistech** [comment]
What a race!! :exploding_head: 

I can't reproduce the error you get during the upload... I'd recommend to close all the tabs of your browser and restart the app.

I didn't expected such large tracks, the app had issues to the large number of Digital Elevation Model tiles required to represent the background.

I'll find a way to optimize the memory footprint :eyes:

**Comment by beretor** [comment]
yes. wrote the issue just to be able to flex ;-)

joke aside, you were correct about restarting chrome. now it uploads but indeed crashes while trying to display it

just for reference:

`A child process terminated abruptly, the process pool is not usable anymore
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/nicegui/events.py", line 417, in wait_for_result
    await result
  File "/workspaces/pretty-gpx/pretty_gpx/main.py", line 55, in on_multi_upload
    await _on_upload(contents, msg)
  File "/workspaces/pretty-gpx/pretty_gpx/main.py", line 39, in _on_upload
    cache = await run.cpu_bound(process_files, contents, PAPER_SIZES[safe(paper_size_mode_toggle.value)])
  File "/usr/local/lib/python3.10/site-packages/nicegui/run.py", line 69, in cpu_bound
    return await _run(process_pool, safe_callback, callback, *args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/nicegui/run.py", line 52, in _run
    return await loop.run_in_executor(executor, partial(callback, *args, **kwargs))
  File "uvloop/loop.pyx", line 2729, in uvloop.loop.Loop.run_in_executor
  File "/usr/local/lib/python3.10/concurrent/futures/process.py", line 720, in submit
    raise BrokenProcessPool(self._broken)
concurrent.futures.process.BrokenProcessPool: A child process terminated abruptly, the process pool is not usable anymore`

**Comment by fricour** [comment]
Hi,

Your project looks really impressive, and I'm excited about the prospect of creating posters and experimenting with additional colors!

However, I've encountered some issues with file uploads that are similar to what have been described. I obtained several GPX files from Komoot, all under 20MB in size, but I've run into a problem as shown in this image:

![image](https://github.com/user-attachments/assets/ed330ee3-36d5-4a10-aa94-2bdce65c4e42)

Essentially, when I click the upload button, nothing happens. I've attempted to simplify my GPX files, but this didn't resolve the issue. I've tested this on Brave, Chrome, and Firefox, each time using only a single browser tab, but the problem persists.
Interestingly, I also tried using some of the GPX files from the example folder you provided, and I encountered the same error. It seems that only the default example launched by the app works correctly.

Have you encountered this issue before or do you have any suggestions for troubleshooting?

**Comment by ThomasParistech** [comment]
Hi @fricour !
All the GPX examples should be working fine.
Since the example track is correctly being processed, it can be related to the way I run tasks in the background. It works fine on my laptop but I might have introduced a subtle bug when adding the profiler and updating the way we run a job in parallel.

Could you send me some GPX tracks ?
I'll have a look, and I'll try to improve the error handling to better understand what's going on👀

**Comment by ThomasParistech** [comment]
I'll also add a warning to handle huge tracks like the one from @beretor ;)

**Comment by fricour** [comment]
Thanks @ThomasParistech, you should have received some examples in your mailbox.

**Comment by ThomasParistech** [comment]
Too large files will trigger an error notifcation in the gui, without crashing
https://github.com/ThomasParistech/pretty-gpx/pull/30

**Comment by ThomasParistech** [comment]
@fricour 
I can't reproduce your error
But it seems to be related to Matplotlib. According to your screenshot it crashes during the text optimisation when we open a matplotlib context

I've created a new issue
https://github.com/ThomasParistech/pretty-gpx/issues/29
